pkg/certagent/agent.go

```diff
@@ -108,7 +118,7 @@ func (c *CertAgent) RequestCertificate() error {
 	}
 
 	// send CSR to the signer
-	if _, err := c.client.Create(csr); err != nil {
+	if _, err := c.client.Create(csr); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
this will cause the client to accept the certificate from old csr request but `GenerateCSRObject` has replaced the private key on disk with new one...

we should't generate if csr with the expected name already exists.
@csrwng

Reply:
@abhinavdahiya I see, you're right. 
We need to fix the flow though. In the case that we time out waiting for the certificate to be signed, we will fail and try again, generating a new private key, but will never be able to create another CSR because one already exists. Should the CSR name be randomized? (as in using GenerateName?)